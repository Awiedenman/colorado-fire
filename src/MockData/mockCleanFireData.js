/* eslint-disable */
export const mockCleanFireData = [{
  fire_name: "416",
  acres_burned: " 54129 acres",
  image: undefined,
  last_update: " 7/24/2018 10:15:00 PM",
  latitude: "37.4605800000001",
  longitude: "-107.8081",
  verified: true
}]