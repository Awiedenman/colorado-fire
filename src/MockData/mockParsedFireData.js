export const mockParsedFireData = {
  rss: {
    channel: {
      item: [{
        description: {
          _text: " 7/24/2018 10:15:00 PM,  54129 acres, Lat/Lon: 37.4605800000001/-107.8081"
        },
          enclosure: {
            _attributes: {
              
            },
        },
          ['geo:lat']: {_text: "37.4605800000001"
        },
          ['geo:long']: {_text: "-107.8081"
        },
          link:{_text: "https://rmgsc-haws2.cr.usgs.gov/geomac/DynContent/georss/416 .png"
        },
          pubDate:{_text: "Wed, 25 Jul 2018 11:00:11 MDT"
        },
          title:{_text: "416 ,  2018-COSJF-000416"
        }
      }]
    }

  }
}