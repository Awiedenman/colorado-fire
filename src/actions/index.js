export const storeCurrentFireData = (cleanedCurrentFireData) => ({
  type: 'STORE_CURRENT_FIRE_DATA',
  cleanedCurrentFireData
});