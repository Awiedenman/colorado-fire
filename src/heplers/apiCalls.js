var convert = require('xml-js');
 
 export const currentFireRequest = async () => {
   const url = 'https://cors-anywhere.herokuapp.com/https://www.geomac.gov/DynContent/georss/nifc_large_firesW3C.xml';
   try{
     const xmlResponse = await fetch(url);
     const currentFireDataXml = await xmlResponse.text();
     var jsonData = convert.xml2json(
       currentFireDataXml, {
         compact: true,
         spaces: 2
        });
        // console.log(currentFireDataXml)
        // console.log(jsonData)
        return jsonData
      } catch(error){
        throw Error('Couldn\'t retreive the current fire')
      }
 }