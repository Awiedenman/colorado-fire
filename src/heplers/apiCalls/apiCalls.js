var convert = require('xml-js');
 
 export const currentFireRequest = async () => {
   const url = 'https://cors-anywhere.herokuapp.com/https://www.geomac.gov/DynContent/georss/nifc_large_firesW3C.xml';
   try{
     const xmlResponse = await fetch(url);
     const currentFireDataXml = await xmlResponse.text();
     var jsonData = convert.xml2json(
       currentFireDataXml, {
         compact: true,
         spaces: 2
        });
        return jsonData
      } catch(error){
        throw Error('Couldn\'t retreive the current fires list')
      }
 }

 export const currentFireArticlesRequest = async () => {
   const url = 'https://cors-anywhere.herokuapp.com/http://inciweb.nwcg.gov/feeds/rss/articles/';
   try {
     const xmlResponse = await fetch(url);
     const currentFireArticlesXml = await xmlResponse.text();
     var jsonData = convert.xml2json(
       currentFireArticlesXml, {
         compact: true,
         spaces: 2
       });
    //  console.log(currentFireArticlesXml)
    //  console.log(jsonData)
     return jsonData
   } catch (error) {
     throw Error('Couldn\'t retreive the current fire')
   }
 }