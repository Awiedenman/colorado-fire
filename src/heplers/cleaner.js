export const fireDataCleaner = (fireData) => {
  const cleanFireData = fireData.rss.channel.item.reduce((fireObject, fire )=> {
    // console.log(fire)
    // console.log(fire.title._text.split(',')[0]);
    fireObject['name'] = fire.title._text.split(',')[0],
    fireObject['image'] = fire.enclosure._attributes.url
    fireObject['acresBurned'] = fire.description._text.split(',')[1]

    console.log(fireObject)
    return fireObject
  }, {})
}