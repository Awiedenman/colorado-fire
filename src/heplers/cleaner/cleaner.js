export const fireDataCleaner = (fireData) => {
  // console.log(fireData)
  const cleanFireData = fireData.rss.channel.item.reduce((currentFires, fire )=> {
    const fireObject = {};
    fireObject['name'] = fire.title._text.split(',')[0],
    fireObject['image'] = fire.enclosure._attributes.url,
    fireObject['acresBurned'] = fire.description._text.split(',')[1],
    fireObject['lastUpdate'] = fire.description._text.split(',')[0]
    fireObject['latitude'] = fire['geo:lat']._text,
    fireObject['longitude'] = fire['geo:long']._text
    currentFires.push(fireObject)
    return currentFires
  }, [])
  // console.log('out', cleanFireData);
  return cleanFireData
}
