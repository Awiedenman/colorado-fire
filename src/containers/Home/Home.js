import React, {Component} from 'react';
import CurrentFires from '../CurrentFires/CurrentFires';
import { fireDataCleaner } from '../../heplers/cleaner';
import { currentFireRequest } from '../../heplers/apiCalls/apiCalls';
import { storeCurrentFireData } from '../../actions/index';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';


export class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoading: false
    }
  }

  async componentDidMount() {  
    await this.setState({isLoading: true}) 
    const currentFireData = await currentFireRequest();
    await this.setState({isLoading: false})
    const parsedFiresData = JSON.parse(currentFireData)
    const cleanedCurrentFireData = fireDataCleaner(parsedFiresData);
    this.props.storeCurrentFireData(cleanedCurrentFireData)
  }
  style = {
        width: '100%',
        height: '100%'
      }
  render() {
    return(
      <div>
        <CurrentFires
          style = {this.style}/>
      </div>
    )
  }
}

export const mapDispatchToProps = dispatch => ({
  storeCurrentFireData: (cleanedCurrentFireData) => dispatch(storeCurrentFireData(cleanedCurrentFireData))
})
// debugger

export default withRouter(connect(null, mapDispatchToProps)(Home));

