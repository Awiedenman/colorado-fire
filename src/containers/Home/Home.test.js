import React from 'react';
import { shallow } from 'enzyme';
import { Home , mapDispatchToProps } from './Home';
import { storeCurrentFireData } from '../../actions';
import { currentFireRequest } from '../../heplers/apiCalls/apiCalls';
import { fireDataCleaner } from '../../heplers/cleaner/cleaner';
import { mockCleanFireData } from '../../MockData/mockCleanFireData';
import { mockParsedFireData } from '../../MockData/mockParsedFireData';

jest.mock('../../heplers/apiCalls/apiCalls', () => ({
  currentFireRequest: jest.fn().mockImplementation(() => Promise.resolve(mockParsedFireData))
}))

describe('Home', () => {
  let wrapper;
  const mockStoreCurrentFireData = jest.fn(); 

  describe('componentDidMount', () => {
    beforeEach(()=> {
      wrapper = shallow(
      <Home
        storeCurrentFireData = {mockStoreCurrentFireData}
      />)
    })

    test('should fetch initial fire data on page load', async () => {
      await expect(mockStoreCurrentFireData).toHaveBeenCalled();
    })
  })
  
  describe('mapDispatchToProps', () => {
    test('should call dispatch when storeCurrentFireData is called', () => {
      const mockDispatch = jest.fn();
      // const mockCleanFireData = mockCleanFireData

      const actionToDispatch = storeCurrentFireData(mockCleanFireData);
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.storeCurrentFireData(mockCleanFireData)

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })
  })
  
  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})

