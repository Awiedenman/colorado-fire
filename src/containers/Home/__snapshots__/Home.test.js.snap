// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    storeCurrentFireData={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Connect(Wrapper)
        style={
                Object {
                        "height": "100%",
                        "width": "100%",
                      }
        }
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "style": Object {
          "height": "100%",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Connect(Wrapper)
          style={
                    Object {
                              "height": "100%",
                              "width": "100%",
                            }
          }
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "style": Object {
            "height": "100%",
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
